ITMO_FS/ensembles/measure_based/WeightBased.py

```diff
@@ -1,85 +1,51 @@
 from collections import defaultdict
 
 from numpy import ndarray, ones
-from sklearn.base import TransformerMixin
+from sklearn.base import clone
 
 from ITMO_FS.utils.data_check import *
+from ...utils import BaseTransformer
 from .fusion_functions import *
 
 
-class WeightBased(TransformerMixin):
-    __filters = []
+class WeightBased(BaseTransformer):
 
-    def __init__(self, filters):
+    def __init__(self, filters, cutting_rule, fusion_function=weight_fusion, weights=None):
         """
         TODO comments
         :param filters:
         """
-        check_filters(filters)
-        self.__filters = filters
-        self.selected_features = None
-        self.feature_scores = None
+        self.filters = filters
+        self.cutting_rule = cutting_rule
+        self.fusion_function = fusion_function
+        self.weights = weights
+
+    def get_score(self, X, y):
+        self.feature_scores_ = defaultdict(list)
+        for __filter in self.filters:
+            _filter = clone(__filter)
+            _filter.fit(X, y, store_scores=True)
+            _min = min(_filter.feature_scores_.values())
+            _max = max(_filter.feature_scores_.values())
+            for key, value in _filter.feature_scores_.items():
+                self.feature_scores_[key].append((value - _min) / (_max - _min))
+        return self.feature_scores_
 
     def __len__(self):
-        return len(self.__filters)
+        return len(self.filters)
 
-    def score(self, X, y, feature_names=None):
-        self.feature_scores = defaultdict(list)
-        for _filter in self.__filters:
-            _filter.fit(X, y, feature_names=feature_names, store_scores=True)
-            _min = min(_filter.feature_scores.values())
-            _max = max(_filter.feature_scores.values())
-            for key, value in _filter.feature_scores.items():
-                self.feature_scores[key].append((value - _min) / (_max - _min))
-        return self.feature_scores
-
-    def fit(self, X, y, feature_names=None):
+    def _fit(self, X, y):
         """
         TODO comments
         :param X:
         :param y:
         :param feature_names:
         :return:
         """
-        feature_names = generate_features(X, feature_names)
-        check_shapes(X, y)
-        # check_features(features_names)
-        _feature_names = feature_names
-        _X = X
-        _y = y
-        self.feature_scores = self.score(_X, _y, _feature_names)
-
-    def transform(self, x, cutting_rule, fusion_function=weight_fusion, weights=None):
```

Comment:
Why did we drop transform?

Reply:
transform exists in BaseTransformer and is inherited